--- origsrc/fxscintilla-1.71/config/acinclude.m4	2006-09-25 16:25:28.000000000 -0500
+++ src/fxscintilla-1.71/config/acinclude.m4	2006-11-05 20:16:51.483000000 -0600
@@ -20,7 +20,7 @@
 LIBS=-lFOX$1
 AC_LANG_CPLUSPLUS
 
-if test x"$HAVE_CYGWIN" = xyes -o x"$HAVE_MINGW32" = xyes; then
+if test x"$HAVE_MINGW32" = xyes; then
   LDFLAGS="$LDFLAGS -mwindows"
 	if test x"$HAVE_CYGWIN" = xyes; then
 		CPPFLAGS="$CPPFLAGS -DWIN32"
@@ -73,7 +73,6 @@
 #
 AC_REQUIRE([AC_CANONICAL_HOST])[]dnl
 case $host_os in
-  *cygwin* ) HAVE_CYGWIN=yes;;
   *mingw32* ) HAVE_MINGW32=yes;;
          * ) ;;
 esac
--- origsrc/fxscintilla-1.71/configure.in	2006-09-26 16:13:08.000000000 -0500
+++ src/fxscintilla-1.71/configure.in	2006-11-05 20:31:35.644000000 -0600
@@ -4,18 +4,13 @@
 dnl Check Cygwin and MinGW32
 AC_CANONICAL_HOST
 case $host_os in
-	*cygwin* )  HAVE_CYGWIN=yes;;
 	*mingw32* ) HAVE_MINGW32=yes;;
 	* ) ;;
 esac
 
-AM_CONDITIONAL(HAVE_CYGWIN, test x"$HAVE_CYGWIN" = xyes)
 AM_CONDITIONAL(HAVE_MINGW32, test x"$HAVE_MINGW32" = xyes)
 
 dnl Using libtool
-if test x"$HAVE_CYGWIN" = xyes -o x"$HAVE_MINGW32" = xyes; then
-  AC_DISABLE_SHARED
-fi
 AM_PROG_LIBTOOL
 
 dnl Must have a C++ compiler
--- origsrc/fxscintilla-1.71/scintilla/src/Makefile.am	2006-09-25 15:15:17.000000000 -0500
+++ src/fxscintilla-1.71/scintilla/src/Makefile.am	2006-11-05 20:18:31.592375000 -0600
@@ -1,12 +1,6 @@
 INCLUDES = -I$(top_srcdir)/include -I$(top_srcdir)/scintilla/include
 AM_CXXFLAGS = -W -Wall -DFOX
 
-## Cygwin
-if HAVE_CYGWIN
-AM_CXXFLAGS += -DWIN32
-AM_LDFLAGS = -mwindows
-endif # Cygwin
-
 ## MinGW32
 if HAVE_MINGW32
 AM_LDFLAGS = -mwindows
--- origsrc/fxscintilla-1.71/src/Makefile.am	2006-09-25 16:00:18.000000000 -0500
+++ src/fxscintilla-1.71/src/Makefile.am	2006-11-05 20:24:23.493000000 -0600
@@ -6,12 +6,6 @@
 
 AM_CXXFLAGS = -W -Wall -DFOX
 
-## Cygwin
-if HAVE_CYGWIN
-AM_CXXFLAGS += -DWIN32
-AM_LDFLAGS = -mwindows
-endif # Cygwin
-
 ## MinGW32
 if HAVE_MINGW32
 AM_LDFLAGS = -mwindows
--- origsrc/fxscintilla-1.71/src/PlatFOX.cxx	2006-10-13 16:03:34.000000000 -0500
+++ src/fxscintilla-1.71/src/PlatFOX.cxx	2006-11-05 20:19:57.993000000 -0600
@@ -32,7 +32,6 @@
 
 #if !defined(WIN32) || defined(__CYGWIN__)
 # if defined(__CYGWIN__)
-#  include <windows.h>
 #  ifdef PIC
 #   define FOXDLL
 #  endif
--- origsrc/fxscintilla-1.71/src/ScintillaFOX.cxx	2006-09-25 16:00:18.000000000 -0500
+++ src/fxscintilla-1.71/src/ScintillaFOX.cxx	2006-11-05 20:19:57.993000000 -0600
@@ -34,7 +34,6 @@
 
 #if !defined(WIN32) || defined(__CYGWIN__)
 # if defined(__CYGWIN__)
-#  include <windows.h>
 #  ifdef PIC
 #   define FOXDLL
 #  endif
@@ -89,7 +88,7 @@
 #include "SString.h"
 #include "ScintillaBase.h"
 
-#ifdef WIN32
+#if defined(WIN32) || defined(__CYGWIN__)
 #ifdef FOXDLL
 #undef FXAPI
 #define FXAPI __declspec(dllexport)
--- origsrc/fxscintilla-1.71/tests/Makefile.am	2006-09-25 16:00:18.000000000 -0500
+++ src/fxscintilla-1.71/tests/Makefile.am	2006-11-05 20:18:13.061125000 -0600
@@ -15,12 +15,6 @@
 
 AM_CXXFLAGS = $(PTHREAD_CFLAGS) -W -Wall -DFOX -DSCI_LEXER
 
-## Cygwin
-if HAVE_CYGWIN
-AM_CXXFLAGS += -DWIN32
-AM_LDFLAGS = -mwindows
-endif # Cygwin
-
 ## MinGW32
 if HAVE_MINGW32
 AM_LDFLAGS = -mwindows
